### PR TITLE
Created test scenario for audit logging when retrying.

### PR DESCRIPTION
**Pull request by carlogeenen**

### CODE REVIEW

**Comment by carlogeenen** [comment]
Build fails because of flickering tests, confirmed by Martijn.

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/glue/steps/ws/smartmetering/smartmeteringconfiguration/AuditTrail.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        }
+    }
+
+    /*
+     * it may take some time before the records are to the dev_log_item table,
+     * so we have to poll.
+     */
+    private List<DeviceLogItem> findDeviceLogItems(final String deviceIdentification, final int minExcepted) {
+        int loopCount = 0;
+        while (loopCount++ < 150) {
```
Comment (robindenadel) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/glue/steps/ws/smartmetering/smartmeteringconfiguration/AuditTrail.java]: use the standard polling mechanism, instead of introducing a new one

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/glue/steps/ws/smartmetering/smartmeteringconfiguration/AuditTrail.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    private ServiceEndpoint serviceEndpoint;
+
+    @Then("^the audit trail contains multiple retry log records$")
+    public void theAuditTrailContainsMultipleRetryLogRecords(final Map<String, String> settings) throws Throwable {
+        final Pattern responsePattern = Pattern.compile(XPATH_MATCHER_RETRY_OPERATION);
+
+        final String deviceIdentification = getString(settings, Keys.KEY_DEVICE_IDENTIFICATION,
+                Defaults.DEFAULT_DEVICE_IDENTIFICATION);
+
+        final List<DeviceLogItem> deviceLogItems = this.findDeviceLogItems(deviceIdentification, 2);
```
Comment (robindenadel) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/glue/steps/ws/smartmetering/smartmeteringconfiguration/AuditTrail.java]: create a constant minExpected, instead of using the magic number 2

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/glue/steps/ws/smartmetering/smartmeteringconfiguration/AuditTrail.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        if (null == deviceLogItems) {
+            Assert.fail("DeviceLogItems where not found in the database");
+        }
+
+        for (int i = 0; i < 2; i++) {
```
Comment (robindenadel) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/glue/steps/ws/smartmetering/smartmeteringconfiguration/AuditTrail.java]: use minExpected here as well

File: cucumber-tests-platform-dlms/src/test/resources/features/osgp-adapter-ws-smartmetering/SmartMeteringConfiguration.feature [hunk 4]
```diff
@@ -139,3 +139,13 @@ Feature: SmartMetering Configuration
     When the request for a firmware upgrade is received
     And the upgrade of firmware did not succeed
     Then the message "Upgrade of firmware did not succeed" should be given
+@Skip
```
Comment (robindenadel) [inline cucumber-tests-platform-dlms/src/test/resources/features/osgp-adapter-ws-smartmetering/SmartMeteringConfiguration.feature]: Remove the skop annotation.

Comment (carlogeenen) [inline cucumber-tests-platform-dlms/src/test/resources/features/osgp-adapter-ws-smartmetering/SmartMeteringConfiguration.feature]: Will do after merging and deploying.

**Comment by robindenadel** [comment]
After a new deployment, the scenario 'Try to connect to an unknown ip address' fails

**Comment by carlogeenen** [comment]
retest this please

**Comment by carlogeenen** [comment]
please retest this

**Comment by carlogeenen** [comment]
retest this please

**Comment by carlogeenen** [comment]
retest this please

File: cucumber-tests-platform-dlms/src/test/resources/features/osgp-adapter-ws-smartmetering/Core.feature [hunk 5]
```diff
@@ -0,0 +1,14 @@
+Feature: Core operations
```
Comment (robindenadel) [inline cucumber-tests-platform-dlms/src/test/resources/features/osgp-adapter-ws-smartmetering/Core.feature]: feature should be in osgp-adapter-ws-smartmetering

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/glue/steps/ws/smartmetering/smartmeteringconfiguration/AuditTrail.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        final List<DeviceLogItem> deviceLogItems = this.deviceLogItemRepository
+                .findByDeviceIdentification(deviceIdentification, new PageRequest(0, 2)).getContent();
+
+        for (int i = 0; i < MULTIPLE_RETRY_COUNT; i++) {
+            final DeviceLogItem item = deviceLogItems.get(i);
+            LOGGER.info("CreationTime: {}", item.getCreationTime().toString());
+            LOGGER.info("DecodedMessage: {}", item.getDecodedMessage());
+
+            final Matcher responseMatcher = responsePattern.matcher(item.getDecodedMessage());
+            assertTrue("A retrying operation is logged.", responseMatcher.find());
```
Comment (robindenadel) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/glue/steps/ws/smartmetering/smartmeteringconfiguration/AuditTrail.java]: Change the message to an error message

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/glue/steps/ws/smartmetering/smartmeteringconfiguration/AuditTrail.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+import com.alliander.osgp.cucumber.platform.Keys;
+import com.alliander.osgp.cucumber.platform.dlms.glue.steps.ws.smartmetering.SmartMeteringStepsBase;
+import com.alliander.osgp.cucumber.platform.dlms.support.ResponseNotifier;
+import com.alliander.osgp.cucumber.platform.dlms.support.ServiceEndpoint;
+import com.alliander.osgp.logging.domain.entities.DeviceLogItem;
+import com.alliander.osgp.logging.domain.repositories.DeviceLogItemRepository;
+
+import cucumber.api.java.en.Then;
+
+public class AuditTrail extends SmartMeteringStepsBase {
```
Comment (robindenadel) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/glue/steps/ws/smartmetering/smartmeteringconfiguration/AuditTrail.java]: remove unused fields

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/glue/steps/ws/smartmetering/smartmeteringconfiguration/AuditTrail.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2016 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package com.alliander.osgp.cucumber.platform.dlms.glue.steps.ws.smartmetering.smartmeteringconfiguration;
```
Comment (robindenadel) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/glue/steps/ws/smartmetering/smartmeteringconfiguration/AuditTrail.java]: Change package according to the location of the feature file

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/glue/steps/ws/smartmetering/smartmeteringconfiguration/AuditTrail.java [hunk 9]
```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2016 Smart Society Services B.V.
```
Comment (robindenadel) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/glue/steps/ws/smartmetering/smartmeteringconfiguration/AuditTrail.java]: 2016 -> 2017

File: cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/AcceptanceTests.java [hunk 10]
```diff
@@ -14,11 +14,11 @@
 import cucumber.api.junit.Cucumber;
 
 @RunWith(Cucumber.class)
-@CucumberOptions(features = "src/test/resources/features", tags = { "~@Skip" }, glue = {
+@CucumberOptions(features = "src/test/resources/features", tags = { "@test" }, glue = {
```
Comment (robindenadel) [inline cucumber-tests-platform-dlms/src/test/java/com/alliander/osgp/cucumber/platform/dlms/AcceptanceTests.java]: revert these changes

File: cucumber-tests-platform-dlms/src/test/resources/features/osgp-adapter-ws-core/AuditTrail.feature [hunk 11]
```diff
@@ -2,7 +2,7 @@ Feature: Core operations
   As a grid operator
   I want to log each retry in the audit trail
   So I get better information the number of retries necessary
-
+@test
```
Comment (robindenadel) [inline cucumber-tests-platform-dlms/src/test/resources/features/osgp-adapter-ws-core/AuditTrail.feature]: remove this tag
